cmd/linkcheck/links.go

```diff
@@ -129,7 +129,9 @@ func newWalkFunc(invalidLink *bool, client *http.Client) filepath.WalkFunc {
 				if err != nil {
 					break
 				}
-				if resp.StatusCode == http.StatusTooManyRequests {
+				// This header is used in 301, 429 and 503.
+				// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After
+				if resp.StatusCode == http.StatusTooManyRequests || resp.StatusCode == http.StatusMovedPermanently || resp.StatusCode == http.StatusServiceUnavailable {
```

Comment:
Go client will follow redirects automatically so the 301 check is probably unnecessary.

Reply:
Thanks a lot, Fixed.